json_spec/README.md

```diff
@@ -1,16 +1,2589 @@
+        <td><strong>Object</strong></td>
+        <td><strong>Description</strong></td>
+    </tr>
+    <tr>
+        <td><a href="#schemaversion"><em>SchemaVersionArtifact</em></a></td>
+        <td>An informational element conveying details about the data to be exported, primarily the version of the emitted data to follow. </td>
+    </tr>
+    <tr>
+        <td><a href="#test-run-artifacts"><em>TestRunArtifact</em></a></td>
+        <td>An informational element conveying details about a top level <a href="#def-testrun"><strong>test run</strong></a>. A <a href="#def-testrun"><strong>test run</strong></a> consists of zero or more <a href="#def-teststep"><strong>test steps</strong></a>.</td>
+    </tr>
+    <tr>
+        <td><a href="#test-step-artifacts"><em>TestStepArtifact</em></a></td>
+        <td>An informational element conveying details about a <a href="#def-teststep"><strong>test step</strong></a> which was executed as part of a <a href="#def-testrun"><strong>test run</strong></a>.</td>
+    </tr>
+</table>
+
+
+### OutputArtifact Attributes
+
+All top level [_OutputArtifacts_](#outputartifacts) contain the following attributes in addition to one artifact message which may be a [_SchemaVersionArtifact_](#schemaversion), [_TestRunArtifact_](#test-run-artifacts), or [_TestStepArtifact_](#test-step-artifacts) message.
+
+<table>
+    <tr>
+        <td><strong>Attribute</strong></td>
+        <td><strong>Type</strong></td>
+        <td><strong>Required</strong></td>
+        <td><strong>Description</strong></td>
+    </tr>
+    <tr>
+        <td><em>sequenceNumber</em></td>
+        <td>number (integer)</td>
+        <td><strong>Yes</strong></td>
+        <td>A monotonically increasing field indicating the order of output returned from a <a href="#def-diagnostic"><strong>diagnostic</strong></a>. A sequence number is valuable for determining if information regarding a <a href="#def-diagnostic"><strong>diagnostic's</strong></a> execution has been lost in transmission or transformation and for re-sequencing possibly out of order transmissions.</td>
+    </tr>
+    <tr>
+        <td><em>timestamp</em></td>
+        <td>string</td>
+        <td><strong>Yes</strong></td>
+        <td>A timestamp indicating when the information was generated by the <a href="#def-diagnostic"><strong>diagnostic</strong></a> returned in an ISO date-time format (see <a href="#datetime-strings">Date/Time strings</a> for details).</td>
+    </tr>
+    <tr>
+        <td><strong><em>&lt;artifact&gt;</em></strong></td>
+        <td>JSON: object<br/>Message: various</td>
+        <td><strong>Yes</strong></td>
+        <td>A <a href="#schemaversion"><em>SchemaVersionArtifact</em></a>, <a href="#test-run-artifacts"><em>TestRunArtifact</em></a>, or <a href="#test-step-artifacts"><em>TestStepArtifact</em></a> message.</td>
+    </tr>
+</table>
+
+A diagram of the complete result structure of this [**specification**](#def-spec) for data messages is shown below.
+
+---
+
+# TODO:
```

Comment:
Would remove the TODO.

Reply:
ok, we'll add the diagram later as discussed

Reply:
done. please check/resolve